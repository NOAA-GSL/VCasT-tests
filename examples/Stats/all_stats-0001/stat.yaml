# Date and Time Settings
start_date: "2024-04-01_02:00:00"  # Start date of the data processing
end_date: "2024-04-01_02:00:00"    # End date of the data processing
interval_hours: "1"                # Interval for processing (e.g., hourly, 6-hourly)

start_lead_time: 1
end_lead_time: 1
interval_lead_time: 1

# Forecast Configuration
fcst_file_template: "../input_files/HRRR/hrrr.{year}{month}{day}{hour}.wrfprsf{lead_time}.T2M.grib2"  # Reference file template
fcst_var: "2t"                    # Forecast variable name
fcst_level: 2
fcst_type_of_level: "heightAboveGround"
shift: 0                          # Time shift (if any)


# Reference Configuration
ref_file_template: "../input_files/HRRR/hrrr.{valid_year}{valid_month}{valid_day}{valid_hour}.wrfprsf00.T2M.grib2"
ref_var: "2t"  
ref_level: 2
ref_type_of_level: "heightAboveGround" #"heightAboveGround"
               

# Output Configuration
output_dir: "."  # Directory to save outputs
output_filename: "stat.out"  # Output file template

# Statistical Metrics
stat_type: "det"
stat_name:
  - "rmse"
  - "bias"
  - "quantiles"
  - "mae"
  - "gss"
  - "fbias"
  - "pod"
  - "far"
  - "csi"
  - "corr"
  - "stdev"
  - "sr"


threshold:

# Variable Threshold and Radius
var_threshold: 290                  # Threshold value for analysis
var_radius: 2                        # Influence radius for calculations (grid points)

# Grid and Interpolation Settings
interpolation: false               # Whether to interpolate data
target_grid: "../input_files/REF/ref_2024040102_f001.nc"   # Target grid specification

# Parallel Processing
processes: 100                     # Number of processes to run in parallel
