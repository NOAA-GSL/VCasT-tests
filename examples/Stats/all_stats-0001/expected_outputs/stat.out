date	lead_time	rmse	bias	25p	50p	75p	IQR	LW	UW	mae	gss	fbias	pod	far	csi	corr	stdev	sr
2024-04-01 02:00:00	1	0.3606800483424207	0.014733754043505267	-0.23724365234375	-0.04974365234375	0.20025634765625	0.4375	-0.89349365234375	0.85650634765625	0.2715349051848177	0.0	nan	nan	1.0	0.0	0.9988770537429664	7.468718462084811	0.0
